src/mscorlib/src/System/Reflection/Runtime/Assemblies/AssemblyNameHelpers.cs

```diff
@@ -0,0 +1,156 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+/*============================================================
+**
+  Type:  AssemblyNameHelpers
+**
+==============================================================*/
+
+using System;
+using System.Globalization;
+using System.IO;
+using System.Text;
+using System.Collections.Generic;
+
+namespace System.Reflection.Runtime.Assemblies
+{
+    internal static class AssemblyNameHelpers
+    {
+        private const int PUBLIC_KEY_TOKEN_LEN = 8;
+
+        public static String ComputeDisplayName(AssemblyName a)
+        {
+            if (a.Name == String.Empty)
+                throw new FileLoadException();
+
+            StringBuilder sb = new StringBuilder();
+            if (a.Name != null)
+            {
+                sb.AppendQuoted(a.Name);
+            }
+
+            if (a.Version != null)
+            {
+                Version canonicalizedVersion = a.Version.CanonicalizeVersion();
+                if (canonicalizedVersion.Major != ushort.MaxValue)
+                {
+                    sb.Append(", Version=");
+                    sb.Append(a.Version.ToString());
```

Comment:
You need to continue using the canonicalizedVersion to be compatible with the desktop.

The canonicalizedVersion is what truncates the 32-bit Version components into unsigned 16 bit numbers (to match the underlying metadata structures on both N and ECMA and ensure that the AssemblyName is roundtrippable from them.)

A Version component whose lower 16-bits is 0xffff is treated as the "not specified" value and causes it and the numbers that come after it to be dropped from the output (see code sample below) It's probably better not to use Version.ToString() here and implement the logic yourself.

```
        public static void Main()
        {
            AssemblyName a = new AssemblyName("Foo");
            a.Version = new Version(1, 2, 0x1ffff, 4);
            Console.WriteLine("Version.ToString()    = " + a.Version);
            Console.WriteLine("AssemblyName.FullName = " + a.FullName);
        }

```
outputs:
 
```
 Version.ToString()    = 1.2.131071.4
 AssemblyName.FullName = Foo, Version=1.2

```

on the Desktop.